src/Main.java

```diff
@@ -99,29 +86,38 @@ public synchronized void down() {
     }
 
     private void busArrived() {
-        System.out.println("Bus: " + busId + " arrived");
+        System.out.println("Bus: " + busId + " arrived.");
     }
 
     private void busDeparted() {
-        System.out.println("Bus: " + busId + " departed");
+        System.out.println("Bus: " + busId + " departed.");
     }
 
     public void run() {
 
-        Main.mutex.lock();
         Bus.currentBus = this;
```

Comment:
What if another bus came while one bus being boarded? Doesn't it need to be inside the mutex?